qiita_db/user.py

```diff
@@ -475,6 +477,46 @@ def unread_messages(self):
             return qdb.sql_connection.TRN.execute_fetchindex()
 
     # ------- methods ---------
+    def user_artifacts(self, artifact_type=None):
+        """Returns the artifacts owned by the user, grouped by study
+
+        Parameters
+        ----------
+        artifact_type : str, optional
+            The artifact type to retrieve. Default: retrieve all artfact types
+
+        Returns
+        -------
+        dict of {qiita_db.study.Study: list of qiita_db.artifact.Artifact}
+            The artifacts owned by the user
+        """
+        with qdb.sql_connection.TRN:
+            sql_args = [self.id, qiita_config.portal]
+            sql_a_type = ""
+            if artifact_type:
```

Comment:
Sounds good.

On (Mar-28-16|16:22), Antonio Gonzalez wrote:
> > +        """Returns the artifacts owned by the user, grouped by study
> > +
> > +        Parameters
> > +        ----------
> > +        artifact_type : str, optional
> > +            The artifact type to retrieve. Default: retrieve all artfact types
> > +
> > +        Returns
> > +        -------
> > +        dict of {qiita_db.study.Study: list of qiita_db.artifact.Artifact}
> > +            The artifacts owned by the user
> > +        """
> > +        with qdb.sql_connection.TRN:
> > +            sql_args = [self.id, qiita_config.portal]
> > +            sql_a_type = ""
> > +            if artifact_type:
> 
> I think is fine to return an empty dict. This is also happening in other sections of the code.
> 
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1716/files/8365dceee043c29c50ea3a58e61887e96b3f99a2#r57650695